internal/controller/controllers/agent_controller.go

```diff
@@ -1908,18 +1910,36 @@ func (r *AgentReconciler) restoreHostByAgent(ctx context.Context, agent *aiv1bet
 		if err != nil {
 			r.Log.WithError(err).Errorf("Failed to get cluster (name: %s namespace: %s)",
 				agent.Spec.ClusterDeploymentName.Name, agent.Spec.ClusterDeploymentName.Namespace)
-			return err
+			return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, err
 		}
 		clusterID = cluster.ID
 	}
 
+	// Check if a host with the same ID already exists (regardless of namespace)
+	// This prevents duplicate hosts and reports conflicts
+	existingHost, err := r.Installer.GetHostByIdInternal(ctx, agent.Name)
+	if err == nil && existingHost != nil {
+		errMsg := fmt.Sprintf("Host with ID %s already exists in namespace %s. Agent creation conflicts with existing host.",
+			agent.Name, existingHost.KubeKeyNamespace)
+		r.Log.Error(errMsg)
+		return r.updateStatus(ctx, log, agent, origAgent, nil, nil, errors.New(errMsg), false)
```

Comment:
Just about the 'internal' flag (which is an indication whether it's a user error), maybe we can check first if the error is indeed ErrRecordNotFound (which means 'false'). I.e. otherwise, in case of some internal db error, the internal flag should be true.
E.g. see: https://github.com/openshift/assisted-service/blob/a7a1ec4de3e374474dc226febb61fea011a61f3c/internal/host/host.go#L1380